docs/v1/P4Runtime-Spec.mdk

```diff
@@ -3318,11 +3322,12 @@ error code.
 We leverage Protobuf's ability to differentiate between set and unset fields to
```

Comment:
The spec should be backward compatible. However it's unclear to me why returning UNIMPLEMENTED would break backwards compatibility here, so maybe you could clarify with an example? In particular, as far as older clients are concerned, nothing would change. For newer clients, yes the behavior would be different based on whether the server is old (fields ignored) or new but doesn't support the feature (UNIMPLEMENTED returned). But I wouldn't say that it breaks backward compatibility. We have been using this document (https://cloud.google.com/apis/design/compatibility#adding_a_field_to_a_request_message) as a reference IIRC, and all I see there is:

> Adding request fields can be non-breaking, so long as clients that don't specify the field will be treated the same in the new version as in the old version.

TLDR; my opinion here is that because old implementations are not impacted (e.g. an old client), we are not breaking backwards compatibility. New clients may need to handle different behaviors (based on whether the server is old or new), but these are new clients so it should be ok.

Reply:
I agree.
In spirit of that, is it necessary to mention that the counters will be untouched for (unset) INSERT requests ? Can we say the default value is left to the target implementation and is generally 0.
>WriteRequest (INSERT)`
>        * if **unset**: The initial value for all 3 counter entries is the default (0).

Reply:
Thanks for explaining @antoninbas, that makes sense.

> is it necessary to mention that the counters will be untouched for (unset) INSERT requests ? 

A direct meter associates meter state with each table entry. This state does not exist / is not defined before the table entry gets created, as far as v1model or PSA are concerned. So what does it even mean to leave the counters (which are part of the meter state for the table entry) "untouched"? I don't think there is a sensible definition of "untouched"?

> Can we say the default value is left to the target implementation and is generally 0.

Since we already ask targets to support resetting counters to zero, why not do that by default when creating a new entry?

Reply:
I was highlighting this from a compatibility point of view. If targets were setting the meter color counters to a non-zero value (although very unlikely) before, on an INSERT operation, targets complying to the updated spec now need to set it to zero.

It is an unlikely scenario, so I don't mind ignoring this corner case scenario for backward compatibility.